examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+  //   \lambda \delta_{ij} \delta_{kl} +
+  //   \mu (\delta_{ik} \delta_{jl} + \delta_{il} \delta_{jk}).
+  // @f]
+  template <int dim>
+  SymmetricTensor<4, dim> get_stiffness_tensor(const double lambda,
+                                               const double mu)
+  {
+    SymmetricTensor<4, dim> stiffness_tensor;
+    for (unsigned int i = 0; i < dim; ++i)
+      for (unsigned int j = 0; j < dim; ++j)
+        for (unsigned int k = 0; k < dim; ++k)
+          for (unsigned int l = 0; l < dim; ++l)
+            stiffness_tensor[i][j][k][l] =
+              (((i == k) && (j == l) ? mu : 0.0) +
+               ((i == l) && (j == k) ? mu : 0.0) +
+               ((i == j) && (k == l) ? lambda : 0.0));
+    return stiffness_tensor;
+  }
+
+
+
+  // @sect3{The `ElasticWave` class}
+
+  // Next let's declare the main class of this program. Its structure is very
+  // similar to the step-40 tutorial program. The main differences are:
+  // - The sweep over the frequency values.
+  // - We save the stiffness and mass matrices in `quadrature_cache` and
+  //   use them for each frequency step.
+  // - We store the measured energy by the probe for each frequency step in the
+  //   HDF5 file.
+  template <int dim>
+  class ElasticWave
+  {
+  public:
+    ElasticWave(const Parameters<dim> &parameters);
+    ~ElasticWave();
+    void run();
+
+  private:
+    void setup_system();
+    void assemble_system(const double &omega,
+                         const bool &  calculate_quadrature_data);
+    void solve();
+    void set_position_vector();
+    void store_frequency_step_data(const unsigned int &frequency_idx);
+    void output_results();
+
+    //  This is called before every time step to set up a pristine state for the
+    //  cache variables.
+    void setup_quadrature_cache();
+
+    // This function loops over the frequency vector and runs the simulation for
+    // each frequency step.
+    void frequency_sweep();
+
+    // The parameters are stored in this variable.
+    Parameters<dim> parameters;
+
+    MPI_Comm mpi_communicator;
+
+    parallel::distributed::Triangulation<dim> triangulation;
+
+    QGauss<dim>        quadrature_formula;
+    const unsigned int n_q_points;
+
+    // We store the mass and stiffness matrices in this vector.
+    std::vector<QuadratureCache<dim>> quadrature_cache;
+
+    DoFHandler<dim> dof_handler;
+
+    FESystem<dim> fe;
+
+    IndexSet locally_owned_dofs;
+    IndexSet locally_relevant_dofs;
+
+    AffineConstraints<std::complex<double>> constraints;
+
+    LinearAlgebraPETSc::MPI::SparseMatrix system_matrix;
+    LinearAlgebraPETSc::MPI::Vector       locally_relevant_solution;
+    LinearAlgebraPETSc::MPI::Vector       system_rhs;
+
+
+    // This vector contains the range of frequencies that we are going to
+    // simulate
+    std::vector<double> frequency;
+
+    // This vector contains the coordinates $(x,y)$ of the points of the
+    // measurement probe.
+    FullMatrix<double> position;
```

Comment:
I renamed it to probe_positions.

I do agree that `std::vector<Point<dim>>` makes more sense!
Before I use `std::vector<Point<dim>>` I have to add support in HDF5 for that container. I save the variable `probe_positions` in the HDF5 file. How about we leave it for now like this until I add the support in HDF5 for `std::vector<Point<dim>>`. Later I can modify this in another PR.

I think that the best way to save `std::vector<Point<dim>>` in HDF5 is to consider Point<dim> as a compound HDF5 type. The advantages are:
* `std::complex<double>` is already is already a compound type
* All the machinery is already in place. I just have to add the types `Point<1>`, `Point<2>` and `Point<3>` to HDF5::internal::get_hdf5_datatype()
* It will be possible to create 1D, 2D or 3D datasets of Points. In this tutorial I used a 1D dataset of Points. In another I may want to have a 2D probe, in that case I may need a 2D dataset of Points.
* Python understands quite well HDF5 compound types

Reply:
I think that's an ok approach if you're willing to address this in the near future. If you want, you can always open github issues to remind yourself of these things.